src/android/window.kt

```diff
@@ -89,6 +103,10 @@ open class Window (runtime: Runtime, activity: MainActivity) {
         }
       }
     })
+
+    this.activity.get()?.client?.assetManager = runtime.configuration.assetManager
```

Comment:
These 3 lines seem brittle. We should have a method on the `WebViewClient` that sets these. The `WebViewClient` has access to the `WebViewActivity` so it should at least be able to set the `rootDirectory`

Reply:
Can you take a look at the changes I made, I'm not sure if it satisfies your concerns.

`rootDirectory` is on of `MainActivity`, but `WebViewClient` just thinks it's a `WebViewActivity` so it can't access it without a cast.
Also, I am having issues with the order things are initialized in, `Window` owns `getJavaScriptPreloadSource()`, but `Window` is initialized after `WebViewActivity` and consequently after `WebViewClient`.